engine/crates/engine-v2/src/execution/planner/mod.rs

```diff
@@ -1,131 +1,217 @@
 mod builder;
 mod query_modifier;
 
-use std::sync::Arc;
+use std::{mem::take, sync::Arc};
 
+use builder::ExecutionBuilder;
+use id_newtypes::IdRange;
 use itertools::Itertools;
 
-use builder::RequirementsBuildContext;
-
 use crate::{
-    execution::{ExecutionPlan, ExecutionPlanId, PlanningResult, PreExecutionContext},
+    execution::{ExecutionPlan, ExecutionPlanId, PlanningResult, PreExecutionContext, ResponseModifierExecutorId},
     operation::{FieldId, LogicalPlanId, PreparedOperation, Variables},
     response::{ResponseViewSelection, ResponseViews},
     sources::PreparedExecutor,
     utils::BufferPool,
     Runtime,
 };
 
-use super::{header_rule::create_subgraph_headers_with_rules, ExecutableOperation};
+use super::{header_rule::create_subgraph_headers_with_rules, ExecutableOperation, ResponseModifierExecutor};
 
 pub(super) struct ExecutionPlanner<'ctx, 'op, R: Runtime> {
```

Comment:
Does this even need to be public?  Looks like it's only used inside the plan function

Reply:
The `ResponseModifierExecutor` could be `pub(in crate::execution)` currently it's `pub(crate)`

Reply:
I'm talking about ExecutionPlanner - it's `pub(super)` but appears to be both constructed and dropped inside this module.  Seemed like an implementation detail of this module.  Unless I'm missing something

Reply:
ah right, leftover you're right